### PR TITLE
Split code to separate JS file for Polymer 3 compatibility

### PR DESCRIPTION
**Pull request by web-padawan**
The JS file is supposed to be loaded by dynamically created `<script>` tag
See also vaadin/vaadin-development-mode-detector#8

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-usage-statistics/11)

### CODE REVIEW

File: vaadin-usage-statistics.js [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+      const info = 'This request contains usage statistics gathered from the application running in development mode. \n\nStatistics gathering is automatically disabled and excluded from production builds.\n\nFor details and to opt-out, see https://github.com/vaadin/vaadin-usage-statistics.\n\n\n\n';
+      var self = this;
+      this.sender.send(info + JSON.stringify(data), function () {
+        // Revert the 'month processed' flag
+        localStorage.setItem(UsageStatistics.monthProcessedKey, self.currentMonth() - 1);
+      });
+    }
+  }
+
+  window.StatisticsGatherer = StatisticsGatherer;
```
Comment (limonte) [inline vaadin-usage-statistics.js]: can we somehow avoid `window` pollution?

**Comment by web-padawan** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-usage-statistics.js, line 321 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-usage-statistics/11#-L7y11oqfu4pFmO9ZIEf-r1-321:-L7y4uxsADbaSuQ0JuhH:b-j2y1ux) ([raw file](https://github.com/vaadin/vaadin-usage-statistics/blob/865dcc8523f374f936ea5df61139f9198c201973/vaadin-usage-statistics.js#L321)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

can we somehow avoid `window` pollution?
</blockquote></details>

This is currently used is tests, that was the only reason to keep it here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-usage-statistics/11)*

**Comment by web-padawan** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-usage-statistics.js, line 321 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-usage-statistics/11#-L7y11oqfu4pFmO9ZIEf-r1-321:-L7y80VeUjB1usZZksEB:b-rjo7ua) ([raw file](https://github.com/vaadin/vaadin-usage-statistics/blob/865dcc8523f374f936ea5df61139f9198c201973/vaadin-usage-statistics.js#L321)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

This is currently used is tests, that was the only reason to keep it here.
</blockquote></details>

Updated the code to use `Vaadin` namespace. Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-usage-statistics/11)*

**Comment by limonte** [comment]
:lgtm:

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-usage-statistics/11#-:-L7y8EYBljtUTHIPWFv0:bnfp4nl)*

**Comment by limonte** [comment]
Oh, and maybe the `@vaadin/` scope should be added in this PR:
https://github.com/vaadin/vaadin-usage-statistics/blob/master/package.json#L2

**Comment by web-padawan** [comment]
> Oh, and maybe the @vaadin/ scope should be added in this PR:

Done.

---

Review status: 0 of 4 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-usage-statistics/11#-:-L7yGqgwYLEl595aoKie:b-w6n6i5)*

**Comment by platosha** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/vaadin/vaadin-usage-statistics/11#-:-L7yHJcE12z3-I3trN2R:bnfp4nl)*

**Comment by web-padawan** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-usage-statistics/11)*

**Comment by web-padawan** [comment]
Reviewed 1 of 2 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, 1 of 1 files at r4.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-usage-statistics/11)*

**Pull request merged**